polaris-service/src/main/java/org/apache/polaris/service/PolarisApplication.java

```diff
@@ -261,10 +261,7 @@ public void run(PolarisApplicationConfig configuration, Environment environment)
         .addMappingForUrlPatterns(EnumSet.of(DispatcherType.REQUEST), true, "/*");
 
     if (configuration.getRateLimiter() != null) {
-      environment
-          .servlets()
-          .addFilter("ratelimiter", new RateLimiterFilter(configuration.getRateLimiter()))
-          .addMappingForUrlPatterns(EnumSet.of(DispatcherType.REQUEST), true, "/*");
+      environment.jersey().register(new RateLimiterFilter(configuration.getRateLimiter()));
```

Comment:
This seems like a semantic change unrelated to the metrics. If I'm reading this correctly, this changes the rate limiter to only apply to Jersey resources

Reply:
Yes because TimedApplicationEventListener, which is what emits metrics on requests, depends on Jersey events.

Reply:
I get that, but my concern is whether this actually changes the behavior of the system by rate-limiting less things now

Reply:
(Please fact check me as I only started working with Java web services recently)

I think all of our endpoints are in Jersey, so it shouldn't change the scope, but at a minimum it does push rate limiting further into the request handling chain.

The fundamental problem seems to be that only Jersey knows how to map a request path to its handling method (and thus its metric name). If that mapping was available everywhere, TimedApplicationEventListener could be re-written as a top level filter.

Reply:
Semantically, there is a difference, as ServletFilters are applied before Jersey Filters. Practically, however, it shouldn't have an impact. I think the only real dependency the RateLimiter might have is that the RealmContext is set in case there are different rate limits per realm. Since the `ContextResolverFilter` still applies first, we shouldn't see a practical difference.